### PR TITLE
support double quant for weight-only

### PR DESCRIPTION
**Pull request by xin3he**
## Type of Change

new feature

## Description

popular repos, like bitsandbytes, llama.cpp, provides double quant on scale to improve the compression ratio of weight-only model. Theoretically, double quant will quantize the scales of several blocks and use a hyper scale and a hyper zeropoint to recover it.

Using below args in RTNWeightQuantConfig to set double quant for scales
- double_quant_dtype
- double_quant_bits
- double_quant_sym
- double_quant_group_size

Demo code can be found in UT.

## Expected Behavior & Potential Risk

UT pass

## How has this PR been tested?

local

## Dependency Change?

N/A

### CODE REVIEW

File: neural_compressor/torch/quantization/config.py [hunk 1]
```diff
@@ -100,6 +108,10 @@ def __init__(
         self.enable_mse_search = enable_mse_search
         self.group_dim = group_dim
         self.return_int = return_int
+        self.double_quant_bits = double_quant_bits
+        self.double_quant_dtype = double_quant_dtype
+        self.double_quant_sym = double_quant_sym
+        self.double_quant_group_size = double_quant_group_size
```
Comment (yiliu30) [inline neural_compressor/torch/quantization/config.py:114]: Please add the docstring accordingly.

File: neural_compressor/torch/quantization/config.py [hunk 2]
```diff
@@ -76,6 +80,10 @@ def __init__(
         enable_mse_search: bool = False,
         group_dim: int = 1,
         return_int: bool = False,
+        double_quant_dtype: str = "fp32",
```
Comment (yiliu30) [inline neural_compressor/torch/quantization/config.py:83]: Please register the supported options here: https://github.com/intel/neural-compressor/blob/9150181bb2ab71201fbdb052fbcaa2aba18a090a/neural_compressor/torch/quantization/config.py#L117-L125

**Review by yiliu30 — APPROVED** [review]

File: neural_compressor/torch/algorithms/weight_only/gptq.py [hunk 3]
```diff
@@ -726,7 +754,36 @@ def configure(self, bits, perchannel=False, sym=True, mse=False, norm=2.4, grid=
     def find_params(self, x, weight=False):
         dev = x.device
         self.maxq = self.maxq.to(dev)
-
+        if self.wdtype != "int":
+            from .rtn import quant_weight
```
[the remark this answers is not in the document]
Comment (xin3he) [reply neural_compressor/torch/algorithms/weight_only/gptq.py:761]: Good point, I think we don't need a util.py for weight-only, In my previous design, weight_only.py(RTN) is the basic utility for all algos, like awq, gptq.

File: neural_compressor/torch/algorithms/weight_only_algos.py [hunk 4]
```diff
@@ -31,10 +31,12 @@ def rtn_quantize_entry(
     model: torch.nn.Module, configs_mapping: Dict[Tuple[str, callable], RTNWeightQuantConfig], *args, **kwargs
 ) -> torch.nn.Module:
     """The main entry to apply rtn quantization."""
-    from neural_compressor.torch.algorithms.rtn import apply_rtn_on_single_module
+    from .weight_only.rtn import apply_rtn_on_single_module
```
[the remark this answers is not in the document]
Comment (xin3he) [reply neural_compressor/torch/algorithms/weight_only_algos.py:34]: The Weight_only folder is purely for supporting weight_only_algo.py, keeping relative paths for easy importing.

File: neural_compressor/torch/quantization/config.py (hunk 1 above)
Comment (xin3he) [reply neural_compressor/torch/quantization/config.py:121]: fixed

File: neural_compressor/torch/quantization/config.py (hunk 2 above)
Comment (xin3he) [reply neural_compressor/torch/quantization/config.py:86]: done

**Review by changwangss — APPROVED** [review]

**Pull request merged**